extensions/roc-plugin-dredd/src/actions/dredd.js

```diff
@@ -0,0 +1,33 @@
+import { initLog } from 'roc';
+import Dredd from 'dredd';
+
+const log = initLog();
+
+export default ({ context: { config: { settings } } }) => () => {
+    const port = process.env.PORT || settings.runtime.port;
+
+    const dredd = new Dredd({
+        server: `http://localhost:${port}`,
+        options: settings.test.dredd,
+    });
+
+    log.small.info('Testing API using dredd.\n');
+
+    return () => dredd.run((err, stats) => {
+        if (err) {
+            throw err;
+        }
+
+        if (stats.errors) {
+            log.large.error('One or more errors occured while running dredd');
```

Comment:
You do not need to do an early return here, `error` level logging will close down the process manually using `process.exit`. If this is not desired `warning` can be used instead.